crates/krun-server/src/bin/krun-server.rs

```diff
@@ -0,0 +1,27 @@
+use std::process::Command;
+use std::thread;
+
+use anyhow::Result;
+use krun_server::cli_options::options;
+use krun_server::server::start_server;
+
+fn main() -> Result<()> {
+    env_logger::init();
+
+    let options = options().run();
+
+    thread::spawn(move || {
```

Comment:
I believe we might need a channel for graceful shutdown of the server thread... Otherwise no cleanup can happen and the TCP socket is not closed. Is that okay?

Reply:
The moment `options.command` exits, we exit ourselves and the thread and all its associated resources (both kernel and userspace) are released. We would need a sync mechanism if we were spawning a process (fork+execve), but with a thread we can avoid implementing that (in our case) unnecessary boilerplate.

Reply:
Oh, what I mean is to add a `channel`, so that we can send a signal to trigger a graceful shutdown in the server thread, and then we can `join` the server thread to wait for the server thread to fully shutdown before we return from main.

Related:
* https://doc.rust-lang.org/book/ch16-02-message-passing.html
* https://doc.rust-lang.org/book/ch20-03-graceful-shutdown-and-cleanup.html

Reply:
If you don't mind, I can add a commit to do this.

Reply:
I'm aware of that pattern, it's used in almost every device in libkrun. It's also completely unnecessary here.

Reply:
> It's also completely unnecessary here.

I think abruptly stopping the server thread could still cause us issues. For example, the `"OK"` response not being sent despite the command being spawned successfully. I'd have to trace along for any other unintended consequences...

https://github.com/rust-lang/miri/issues/1371#issuecomment-754834840

Reply:
I've just implemented it. To avoid adding new dependencies or complicating the code too much, we're keeping a copy of the RawFd and shutting it down when tearing down things, which causes TcpListener::incoming() to exit, allowing us to wait for and join the server thread successfully.